Ginger/GingerCoreNETUnitTest/AutoPilot/APIModelsCompareTest.cs

```diff
@@ -0,0 +1,181 @@
+﻿using amdocs.ginger.GingerCoreNET;
+using Amdocs.Ginger.Common;
+using Amdocs.Ginger.CoreNET.Repository;
+using Amdocs.Ginger.Repository;
+using Ginger.Run;
+using GingerCoreNET.Application_Models;
+using GingerCoreNETUnitTest.RunTestslib;
+using GingerTestHelper;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace GingerCoreNETUnitTest.AutoPilot
+{
+    /// <summary>
+    /// API Models Comparison Utility Test
+    /// </summary>
+    /// 
+
+    [TestClass]
+    public class APIModelsCompareTest
+    {
+        static GingerRunner mGR = null;
+
+        static List<string> xmlFiles;
+        static List<ApplicationAPIModel> learnedAPIsList;
+        static ObservableList<ApplicationAPIModel> existingAPIsList;
+        static int testExecutedCount = 0;
+
+        [ClassInitialize]
+        public static void ClassInitialize(TestContext TC)
+        {
+            Reporter.ToLog(eLogLevel.DEBUG, "Creating the GingerCoreNET WorkSpace");
+            WorkSpaceEventHandler WSEH = new WorkSpaceEventHandler();
+            WorkSpace.Init(WSEH);
+
+            string TempSolutionFolder = TestResources.GetTestTempFolder(@"Solutions" + Path.DirectorySeparatorChar + "APIModelsComparisonUtilityTest");
+            if (Directory.Exists(TempSolutionFolder))
+            {
+                Directory.Delete(TempSolutionFolder, true);
+            }
+
+            WorkSpace.Instance.SolutionRepository = GingerSolutionRepository.CreateGingerSolutionRepository();
+            WorkSpace.Instance.SolutionRepository.CreateRepository(TempSolutionFolder);
+
+            NewRepositorySerializer RS = new NewRepositorySerializer();
+            NewRepositorySerializer.AddClassesFromAssembly(typeof(ApplicationAPIModel).Assembly);
+            WorkSpace.Instance.SolutionRepository.Open(TempSolutionFolder);
+
+            // Initialize ApplicationAPIModels XML file names to be fetched from TestResources
+            xmlFiles = new List<string>()
+            {
+                @"XML" + Path.DirectorySeparatorChar + "Create_User.Ginger.ApplicationAPIModel.xml",
```

Comment:
please place the  XML's under TestResources/Repository/SampleAPIModels

Reply:
done !